functions/src/webhooks/transcription.ts

```diff
@@ -0,0 +1,22 @@
+import * as functions from "firebase-functions"
+import { AssemblyAI } from "assemblyai"
+import { db } from "../firebase"
+
+const assembly = new AssemblyAI({
+  apiKey: process.env.ASSEMBLY_API_KEY ? process.env.ASSEMBLY_API_KEY : ""
+})
+
+export const transcription = functions.https.onRequest(async (req, res) => {
+  console.log(req.body)
+  if (req.body.status === "completed") {
+    const transcript = await assembly.transcripts.get(req.body.transcript_id)
+    if (transcript) {
+      try {
+        await db.collection("transcriptions").doc(transcript.id).set(transcript)
```

Comment:
Not a blocker for testing (especially since we're still debating the final cut of which fields we actually want to save to our backend and their structure), but let's make a note to revisit the indexes on this collection before signing off on the V1 here - we'll likely want to set up [single field exemptions ](https://firebase.google.com/docs/firestore/query-data/index-overview#indexing_best_practices)for any large string/array fields.